### PR TITLE
AutomationPolicySensorDefinition

### PR DESCRIPTION
**Pull request by sryza**

### CODE REVIEW

File: python_modules/dagster/dagster/_core/definitions/automation_policy_sensor_definition.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+from dagster._annotations import experimental
+from dagster._core.definitions.asset_selection import CoercibleToAssetSelection
+
+from .asset_selection import AssetSelection
+from .sensor_definition import DefaultSensorStatus, SensorDefinition
+from .utils import check_valid_name, validate_tags
+
+
+@experimental
+class AutomationPolicySensorDefinition(SensorDefinition):
```
Comment (gibsondan) [inline python_modules/dagster/dagster/_core/definitions/automation_policy_sensor_definition.py:12]: Despite the name, I was not expecting this to subclass SensorDefinition - I don't love that pattern in the other places that we do it, and i feel like SensorDefinition must make some assumptions that these new "sensors" do not make (for example, that there is a user-specified evaluation function)

Comment (gibsondan) [reply python_modules/dagster/dagster/_core/definitions/automation_policy_sensor_definition.py:12]: We would also need to add logic in the sensor daemon to filter out these types of sensors (which means we need to have some way of identifying them from the snapshot)

**Review by gibsondan — DISMISSED** [review]
the one truly blocking change here is needing a way to identify this type of sensor at the snapshot level so that we can treat it differently in the host cloud - i.e. something that is reflected here: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster/dagster/_core/host_representation/external_data.py#L113-L122

I am unsure if that should be a new top-level field on ExternalRepositoryData - but one benefit of making it so is that we wouldn't have to audit all the existing places we fetch all the sensors on the repository (graphql apis, the sensor daemon, etc.) and make sure that they are handling this new type of sensor correctly

**Comment by sryza** [comment]
After some offline discussion with @gibsondan, I updated the PR to add a new `AUTOMATION_POLICY` value to the `SensorType` enum. This will allow host code to identify automation policy sensors.

**Review by gibsondan — APPROVED** [review]

**Pull request merged**